lib/pbench/cli/agent/commands/tools/list.py

```diff
@@ -33,47 +46,53 @@ def execute(self):
-                        if self.context.with_option:
-                            host_tools[group][path.name] = [
-                                (p, (path / p).read_text()) for p in self.tools(path)
-                            ]
-                        else:
-                            host_tools[group][path.name] = [p for p in self.tools(path)]
+                        host = path.name
+                        tool_info[group][host] = {}
+                        for tool in sorted(self.tools(path)):
+                            tool_info[group][host][tool] = sorted((path / tool).read_text().rstrip('\n').split('\n')) if opts else ""
                 except BadToolGroup:
                     self.logger.error("Bad tool group: %s", group)
-                    # 0.69.9 behavior
-                    return 0
-            if host_tools:
-                for k, v in host_tools.items():
-                    for h, t in v.items():
-                        print("%s: %s %s" % (k, h, t))
+                    return 1
+            if tool_info:
+                self.print_results(tool_info, None, self.context.with_option)
         else:
             # List the groups which include this tool
-            group_list = []
+            tool = self.context.name
+            found = False
             for group in groups:
-                tg_dir = self.gen_tools_group_dir(group)
-                if not tg_dir.exists():
-                    self.logger.error("bad or missing tool group %s", group)
-                    continue
+                try:
+                    tg_dir = self.gen_tools_group_dir(group)
+                except BadToolGroup:
+                    self.logger.error("Bad tool group: %s", group)
+                    return 1
 
+                tool_info[group] = {}
                 for path in tg_dir.iterdir():
                     # skip files like __label__ and __trigger__
                     if not path.is_dir():
                         continue
+
+                    host = path.name
+                    tool_info[group][host] = {}
                     # Check to see if the tool is in any of the hosts.
-                    if self.context.name in self.tools(path):
-                        group_list.append(group)
-            if group_list:
-                print(
-                    "tool name: %s groups: %s"
-                    % (self.context.name, ", ".join(group_list))
+                    if tool in self.tools(path):
+                        tool_info[group][host][tool] = sorted((path / tool).read_text().rstrip("\n").split('\n')) if opts else ""
+                        found = True
+            if found:
+                self.print_results(tool_info, tool, self.context.with_option)
+            else:
+                self.logger.error(
+                    "Tool does not exist in any group: %s", self.context.name
                 )
+                return 1
```

Comment:
It looks like you're missing the return statement for the case where control executes line 90.  (It probably defaults to `None` which is probably a lot like `0`, and so we don't notice, but we should be returning `0` explicitly for the success case.)

Reply:
Done.